### PR TITLE
Test installed CCF

### PR DESCRIPTION
**Pull request by eddyashton**
This tests the installed CCF package in the CI. Rather than only installing on releases we run the `install.yml` step on all of our "Final" builds, running `ninja install` and producing an install tarball. 

`install.yml` now also runs a simple test of the installed binaries and Python infra, starting a network with `start_network.py`. This is deliberately run outside of the `build` directory, and nested further than most of our tests, to check that we're not using any default paths.

### CODE REVIEW

**Comment by cimetrics** [comment]
test_install@[4638 aka 20200207.11](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4638&view=results) vs master ewma over 30 builds from [4334](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4334&view=results) to [4634](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4634&view=results)
images

File: .azure-pipelines-templates/matrix.yml [hunk 1]
```diff
@@ -35,14 +35,13 @@ parameters:
       cmake_args: ''
     debug:
       cmake_args: '-DCMAKE_BUILD_TYPE=Debug -DBUILD_SMALLBANK=OFF'
-    perf:
-      HTTP:
-        cmake_args: ''
-    release:
+    install:
       FTCP:
```
Comment (achamayou) [inline .azure-pipelines-templates/matrix.yml]: Didn't realise I'd left some FTCP stuff in there.

File: tests/infra/remote.py [hunk 2]
```diff
@@ -550,9 +550,14 @@ def __init__(
...
+        # lib_path may be relative or absolute. The remote implementation should
+        # copy (or symlink) to the target workspace, and then node will be able
+        # to reference the destination file locally in the target workspace.
+        enclave_path = os.path.join(".", os.path.basename(lib_path))
+
         cmd = [
             self.BIN,
-            f"--enclave-file={lib_path}",
+            f"--enclave-file={enclave_path}",
             f"--enclave-type={enclave_type}",
```
Comment (olgavrou) [inline tests/infra/remote.py]: We don't have any tests in the CI that check clients/nodes that run remotely on different machines, so might be worth running a test like to to make sure everything works

File: .azure-pipelines-templates/matrix.yml [hunk 3]
```diff
@@ -35,14 +35,9 @@ parameters:
...
-      HTTP:
-        cmake_args: ''
-    release:
-      FTCP:
-        cmake_args: '-DCMAKE_INSTALL_PREFIX=./install'
-      HTTP:
-        cmake_args: '-DCMAKE_INSTALL_PREFIX=./install'
+    install:
+      install_prefix: '/tmp/ccf-install'
+      cmake_args: '-DCMAKE_INSTALL_PREFIX=/tmp/ccf-install'
```
Comment (jumaffre) [inline .azure-pipelines-templates/matrix.yml]: I guess we cannot do `cmake_args: '-DCMAKE_INSTALL_PREFIX=${{ build.install.install_prefix}}'` ?

File: .azure-pipelines-templates/matrix.yml (hunk 1 above)
Comment (eddyashton) [reply .azure-pipelines-templates/matrix.yml]: Removed now.

File: .azure-pipelines-templates/matrix.yml (hunk 3 above)
Comment (eddyashton) [reply .azure-pipelines-templates/matrix.yml]: No: https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=4648&view=results

Comment (eddyashton) [reply .azure-pipelines-templates/matrix.yml]: I'd prefer to just have the `install_prefix`, remove `install: true`, and then in `common.yml` append the cmake_args with something like

`${{ if(parameters.install_prefix, '-DCMAKE_INSTALL_PREFIX=${{ parameters.install_prefix }}', '' }}`

but it seems this is not possible.

File: tests/infra/remote.py (hunk 2 above)
Comment (eddyashton) [reply tests/infra/remote.py]: Quick test suggests it does, thanks!

**Pull request merged**